alg10-swapPair.js

```diff
@@ -0,0 +1,48 @@
+/* Swapping Two by Two
+Write a function that swaps the first pair (1st and 2nd characters) with the second pair (3rd and 4th characters) for every quadruplet substring.
+Examples
+swapTwo("ABCDEFGH") ➞ "CDABGHEF"
+swapTwo("AABBCCDDEEFF") ➞ "BBAADDCCFFEE"
+swapTwo("munchkins") ➞ "ncmuinhks"
+swapTwo("FFGGHHI") ➞ "GGFFHHI"
+Notes
+Keep leftover strings in the same order.*/
+
+const swapPair = str => {
+  var swappedstr = "";
+  var strLnth = str.length;
+  var noOfIterations = 4 * Math.trunc(strLnth / 4);
+  var remaining = strLnth % 4
+
+  if (str.length < 4) {
+    return str;
+  }
+
+  for ( var i = 0; i < noOfIterations; i+=4) {
+    swappedstr += str[i+2] + str[i+3] + str[i] + str[i+1];
+  }
+  
+  if (remaining > 0) {
+    swappedstr += str.substr(noOfIterations);
```

Comment:
wtf no! use a dang for loop

Reply:
explain to me why this is inefficient. If you cant explain it then DONT USE IT! Gonna spank you for not thinking